nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/QueryDatabaseTable.java

```diff
@@ -197,6 +198,12 @@ public void setup(final ProcessContext context) {
         maxValueProperties = getDefaultMaxValueProperties(context.getProperties());
     }
 
+    @OnStopped
+    public void stop() {
+        // Reset the column type map in case properties change
+        setupComplete.set(false);
```

Comment:
can't we just do the setup in `@OnScheduled` and move the setupComplete flag only to `GenerateTableFetch` or remove it? I think the code would be more straightforward like this. What do you think?

Reply:
That's what it used to do, NIFI-4773 was to do the opposite. It's not recommended to try to connect to external systems in `@OnScheduled` due to timeouts and other possible errors. Instead we moved it into onTrigger but only to be done once.

Reply:
I see, thanks for your explaination